<!DOCTYPE html><html lang="en-us" >

<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  
  
  
  <meta name="generator" content="Wowchemy 5.0.0-beta.1 for Hugo">
  

  

  
  

  
  
  
  
  
    
    
    
  
  

  <meta name="author" content="Hector Roux de Bézieux">

  
  
  
    
  
  <meta name="description" content="Currated list of resources that could be useful to beginners (or not) when learning how dive into RNA-Seq Analysis">

  
  <link rel="alternate" hreflang="en-us" href="http://hector-rdb.org/project/resources/">

  







  




  
  
    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
  

  
  
  
  <meta name="theme-color" content="#2962ff">
  

  
  
  
  <script src="/js/mathjax-config.js"></script>
  

  
  
  
  
    
    
      <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.9.0/css/academicons.min.css" integrity="sha512-W4yqoT1+8NLkinBLBZko+dFB2ZbHsYLDdr50VElllRcNt2Q4/GSs6u71UHKxB7S6JEMCp5Ve4xjh3eGQl/HRvg==" crossorigin="anonymous">
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.14.0/css/all.min.css" integrity="sha256-FMvZuGapsJLjouA6k7Eo2lusoAX9i0ShlWFG6qt7SLc=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.css" integrity="sha256-Vzbj7sDDS/woiFS3uNKo8eIuni59rjyNGtXfstRzStA=" crossorigin="anonymous">

    
    
    
      
    
    
      
      
        
          <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.2.0/styles/github.min.css" crossorigin="anonymous" title="hl-light">
          <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.2.0/styles/dracula.min.css" crossorigin="anonymous" title="hl-dark" disabled>
        
      
    

    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.7.1/leaflet.min.css" integrity="sha512-1xoFisiGdy9nvho8EgXuXvnpR5GAMSjFwp40gSRE3NwdUdIMIKuPa7bqoUhLD0O/5tPNhteAsE5XyyMi5reQVA==" crossorigin="anonymous">
    

    

    
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/lazysizes/5.2.2/lazysizes.min.js" integrity="sha512-TmDwFLhg3UA4ZG0Eb4MIyT1O1Mb+Oww5kFG0uHqXsdbyZz9DcvYQhKpGgNkamAI6h2lGGZq2X8ftOJvF/XjTUg==" crossorigin="anonymous" async></script>
      
    
      

      
      

      
    
      

      
      

      
    
      

      
      

      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml.js" integrity="" crossorigin="anonymous" async></script>
      
    
      

      
      

      
    
      

      
      

      
    

  

  
  
  
    
      
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Montserrat:400,700%7CRoboto:400,400italic,700%7CRoboto+Mono&display=swap">
    
  

  
  
  
  
  
  <link rel="stylesheet" href="/css/wowchemy.5a4eb2f7534142caff5bc52d0b722faa.css">

  




  

  


  
  

  

  <link rel="manifest" href="/index.webmanifest">
  <link rel="icon" type="image/png" href="/images/icon_hua2ec155b4296a9c9791d015323e16eb5_11927_32x32_fill_lanczos_center_2.png">
  <link rel="apple-touch-icon" type="image/png" href="/images/icon_hua2ec155b4296a9c9791d015323e16eb5_11927_192x192_fill_lanczos_center_2.png">

  <link rel="canonical" href="http://hector-rdb.org/project/resources/">

  
  
  
  
  
  
  
    
  
  
  <meta property="twitter:card" content="summary_large_image">
  
  <meta property="og:site_name" content="Hector Roux de Bézieux">
  <meta property="og:url" content="http://hector-rdb.org/project/resources/">
  <meta property="og:title" content="Welcome to the RNA-Seq world | Hector Roux de Bézieux">
  <meta property="og:description" content="Currated list of resources that could be useful to beginners (or not) when learning how dive into RNA-Seq Analysis"><meta property="og:image" content="http://hector-rdb.org/media/icon-192.png">
  <meta property="twitter:image" content="http://hector-rdb.org/media/icon-192.png"><meta property="og:locale" content="en-us">
  
    
      <meta property="article:published_time" content="2018-04-18T00:00:00&#43;00:00">
    
    <meta property="article:modified_time" content="2018-04-18T00:00:00&#43;00:00">
  

  


    









<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "Article",
  "mainEntityOfPage": {
    "@type": "WebPage",
    "@id": "http://hector-rdb.org/project/resources/"
  },
  "headline": "Welcome to the RNA-Seq world",
  
  "datePublished": "2018-04-18T00:00:00Z",
  "dateModified": "2018-04-18T00:00:00Z",
  
  "author": {
    "@type": "Person",
    "name": "Hector Roux de Bézieux"
  },
  
  "publisher": {
    "@type": "Organization",
    "name": "Hector Roux de Bézieux",
    "logo": {
      "@type": "ImageObject",
      "url": "http://hector-rdb.org/images/icon_hua2ec155b4296a9c9791d015323e16eb5_11927_192x192_fill_lanczos_center_2.png"
    }
  },
  "description": "Currated list of resources that could be useful to beginners (or not) when learning how dive into RNA-Seq Analysis"
}
</script>

  

  


  


  





  <title>Welcome to the RNA-Seq world | Hector Roux de Bézieux</title>

</head>


<body id="top" data-spy="scroll" data-offset="70" data-target="#TableOfContents" class="page-wrapper  ">

  
  
  
  
  
  
  
  
  <script src="/js/wowchemy-init.min.1df0392aab543464b23a85146803aaf2.js"></script>

  

<aside class="search-results" id="search">
  <div class="container">
    <section class="search-header">

      <div class="row no-gutters justify-content-between mb-3">
        <div class="col-6">
          <h1>Search</h1>
        </div>
        <div class="col-6 col-search-close">
          <a class="js-search" href="#"><i class="fas fa-times-circle text-muted" aria-hidden="true"></i></a>
        </div>
      </div>

      <div id="search-box">
        
        <input name="q" id="search-query" placeholder="Search..." autocapitalize="off"
        autocomplete="off" autocorrect="off" spellcheck="false" type="search" class="form-control">
        
      </div>

      

    </section>
    <section class="section-search-results">

      <div id="search-hits">
        
      </div>

    </section>
  </div>
</aside>



  <div class="page-header">
    












<nav class="navbar navbar-expand-lg navbar-light compensate-for-scrollbar" id="navbar-main">
  <div class="container">

    
    <div class="d-none d-lg-inline-flex">
      <a class="navbar-brand" href="/">Hector Roux de Bézieux</a>
    </div>
    

    
    <button type="button" class="navbar-toggler" data-toggle="collapse"
            data-target="#navbar-content" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
    <span><i class="fas fa-bars"></i></span>
    </button>
    

    
    <div class="navbar-brand-mobile-wrapper d-inline-flex d-lg-none">
      <a class="navbar-brand" href="/">Hector Roux de Bézieux</a>
    </div>
    

    
    
    <div class="navbar-collapse main-menu-item collapse justify-content-start" id="navbar-content">

      
      <ul class="navbar-nav d-md-inline-flex">
        

        

        
        
        
          
        

        

        
        
        
        

        
          
            
            
          
          
            
            
            
              
            
            
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#about"><span>Home</span></a>
        </li>

        
        

        

        
        
        
          
        

        

        
        
        
        

        
          
            
            
          
          
            
            
            
              
            
            
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#projects"><span>Projects</span></a>
        </li>

        
        

        

        
        
        
          
        

        

        
        
        
        

        
          
            
            
          
          
            
            
            
              
            
            
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#mainPublications"><span>Publications</span></a>
        </li>

        
        

        

        
        
        
          
        

        

        
        
        
        

        
          
            
            
          
          
            
            
            
              
            
            
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#talks"><span>Talks</span></a>
        </li>

        
        

        

        
        
        
          
        

        

        
        
        
        

        
          
            
            
          
          
            
            
            
              
            
            
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#experience"><span>Experience</span></a>
        </li>

        
        

        

        
        
        
          
        

        

        
        
        
        

        
          
            
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/files/cv.pdf"><span>CV</span></a>
        </li>

        
        

      

        
      </ul>
    </div>

    <ul class="nav-icons navbar-nav flex-row ml-auto d-flex pl-md-2">

      
      

      
      
      <li class="nav-item">
        <a class="nav-link js-search" href="#" aria-label="Search"><i class="fas fa-search" aria-hidden="true"></i></a>
      </li>
      

      
      
      <li class="nav-item dropdown theme-dropdown">
        <a href="#" class="nav-link" data-toggle="dropdown" aria-haspopup="true" aria-label="Display preferences">
          <i class="fas fa-moon" aria-hidden="true"></i>
        </a>
        <div class="dropdown-menu">
          <a href="#" class="dropdown-item js-set-theme-light">
            <span>Light</span>
          </a>
          <a href="#" class="dropdown-item js-set-theme-dark">
            <span>Dark</span>
          </a>
          <a href="#" class="dropdown-item js-set-theme-auto">
            <span>Automatic</span>
          </a>
        </div>
      </li>
      

      

    </ul>

  </div>
</nav>


  </div>

  <div class="page-body">
    <article class="article article-project">

  












  

  
  
  
<div class="article-container pt-3">
  <h1>Welcome to the RNA-Seq world</h1>

  

  
    


<div class="article-metadata">

  
  

  
  <span class="article-date">
    
    
      
    
    Apr 18, 2018
  </span>
  

  

  

  
  
  
  
  

  
  

</div>

    














  
</div>



  <div class="article-container">

    <div class="article-style">
      <h1 id="overview">Overview</h1>
<p>Recently, Philippe Boileau joined the <a href="https://www.stat.berkeley.edu/~sandrine/" target="_blank" rel="noopener">Dudoit lab</a> and I had to think on which resources to point him to so that he could start getting up to speed with on the field. Together, we hence build the following list of interesting material. Those fall into 4 broad (and largely overlapping) categories:</p>
<ul>
<li><a href="#Bio">How to produce data with sequencing</a></li>
<li><a href="#Data">How to find, access and pre-process data (including how to use bio-informatics tools)</a></li>
<li><a href="#Stat">How to analyze the data:  statistics</a></li>
<li><a href="#Biostat">How to use all of the above to conduct an analysis (with - of course - a strong emphasis on R tools)</a></li>
<li><a href="#Repro">How to do all those analyses correctly: reproducible research</a></li>
</ul>
<p>I&rsquo;ll therefore try to list resources to help in all subjects. This will be a mix of (scientific) papers, books, package vignette, personal tutorials, video&hellip;</p>
<p>If you find any other relevant material, feel free to put an issue on the <a href="https://github.com/HectorRDB/HectorRDB.github.io/issues" target="_blank" rel="noopener">github repo</a>.</p>
<h1 id="before-we-begin">Before we begin</h1>
<p>The resources below broadly assume that you have at least a basic knowledge of <a href="https://cran.r-project.org/" target="_blank" rel="noopener">R</a>, statistics and biology.</p>
<p>Otherwise, for R, I recommend (and I am far from the only one) Garrett Grolemund &amp; Hadley Wickham&rsquo;s <a href="https://r4ds.had.co.nz/" target="_blank" rel="noopener">R For Data Science</a>.  Additionally, the statistical computing facility at Berkeley host yearly R bootcamp. The material is <a href="https://github.com/berkeley-scf/r-bootcamp-2018" target="_blank" rel="noopener">on Github</a> and are a great way to practice (this is how I really learned R, and I had the change to help in the teaching in 2018).</p>
<p>Biological concept you should be familiar with are (with links to the relevant Wikipedia pages): the <a href="https://en.wikipedia.org/wiki/Central_dogma_of_molecular_biology" target="_blank" rel="noopener">Central Dogma</a>, <a href="https://en.wikipedia.org/wiki/RNA_splicing" target="_blank" rel="noopener">mRNA splicing</a>. Keeping up to date with the current knowledge  on majors diseases is also crucial to .</p>
<p>For statistics, I do not have any material to recommend.</p>
<h1 id="Bio">Sequencing methodology</h1>
<ul>
<li>RNA - Sequencing mostly relies on Illumina Sequencing nowadays but it&rsquo;s useful to learn about other sequencing methods. I feel that for this, videos are the best medium to understand the mechanisms.</li>
</ul>
<p><strong>Sanger sequencing</strong> The first method used for sequening. The video below gives a 3mn tutorial on how it works.

<div style="position: relative; padding-bottom: 56.25%; height: 0; overflow: hidden;">
  <iframe src="https://www.youtube.com/embed/FvHRio1yyhQ" style="position: absolute; top: 0; left: 0; width: 100%; height: 100%; border:0;" allowfullscreen title="YouTube Video"></iframe>
</div>
</p>
<p><strong>Illumina Sequencing</strong> Also called Next Generation Sequencing or Short read sequencing. The best resources I have found come from the Illumina Website

<div style="position: relative; padding-bottom: 56.25%; height: 0; overflow: hidden;">
  <iframe src="https://www.youtube.com/embed/fCd6B5HRaZ8" style="position: absolute; top: 0; left: 0; width: 100%; height: 100%; border:0;" allowfullscreen title="YouTube Video"></iframe>
</div>
</p>
<p><strong>PacBio Sequencing</strong> Long read sequencing, 2$^nd$ generation of NGS. In very, short, longer reads with lower accuracy compared to Illumina reads

<div style="position: relative; padding-bottom: 56.25%; height: 0; overflow: hidden;">
  <iframe src="https://www.youtube.com/embed/NHCJ8PtYCFc" style="position: absolute; top: 0; left: 0; width: 100%; height: 100%; border:0;" allowfullscreen title="YouTube Video"></iframe>
</div>
</p>
<p><strong>Oxford NanoPore sequencing</strong> Even longer reads with even lower quality.
<div style="position: relative; padding-bottom: 56.25%; height: 0; overflow: hidden;">
  <iframe src="https://player.vimeo.com/video/211385238" style="position: absolute; top: 0; left: 0; width: 100%; height: 100%; border:0;" title="vimeo video" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>
</div></p>
<ul>
<li>
<p>Single-cell RNA-Sequencing is, of course, conducted at the single-cell level. The way the cells are pre-processed, separated and sequenced is method-dependent. A good overview of those methods can be found in <a href="https://www.nature.com/articles/nprot.2017.149" target="_blank" rel="noopener">Svensson et al</a>.</p>
</li>
<li>
<p>Many other different types of &lsquo;omics exist: the <a href="https://www.youtube.com/user/joshstarmer/videos" target="_blank" rel="noopener">youtube chanel StatQuest</a> gives a good intro to many of them (as well as clear explanations for many statistical analyses). Some &lsquo;omics that are often collected along RNA-Seq are Chip-Seq and ATAC-Seq.</p>
</li>
</ul>
<h1 id="Data">Prepping the data for analysis</h1>
<h2 id="obtaining-data">Obtaining data</h2>
<p>The R <a href="https://www.bioconductor.org/packages/release/bioc/html/GEOquery.html" target="_blank" rel="noopener">GEOQuery package</a> from <a href="https://www.bioconductor.org/" target="_blank" rel="noopener">Bioconductor</a> allows you to download data and sample information from <a href="https://www.ncbi.nlm.nih.gov" target="_blank" rel="noopener">ncbi</a>. Most papers provide GEO accession numbers as a key to access their data and is usually enough.</p>
<h2 id="pre-processing-data">Pre-processing data</h2>
<p>The sequencing step produces reads at the cell level. The next question is how to map them to a reference genome to obtain transcript or gene expression levels. To obtain this matrix, a few bioinformatic tools exist. One series of tool is <a href="https://academic.oup.com/bioinformatics/article/25/9/1105/203994" target="_blank" rel="noopener">Tophat</a>, which was replaced by <a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-4-r36" target="_blank" rel="noopener">Tophat2</a> and then <a href="https://www.nature.com/articles/nmeth.3317" target="_blank" rel="noopener">HISAT</a>, which was finally upgraded to HISAT2. A popular alternative is the <a href="https://academic.oup.com/bioinformatics/article/29/1/15/272537" target="_blank" rel="noopener">STAR algorithm</a>.</p>
<p>Recently, tools that do not rely on full genome alignment have been proposed, including <a href="https://www.nature.com/articles/nbt.3519" target="_blank" rel="noopener">Kallisto</a> and <a href="https://www.nature.com/articles/nmeth.4197" target="_blank" rel="noopener">Salmon</a>. A recent (not yet per-reviewed comparison) can be found by <a href="https://www.biorxiv.org/content/10.1101/444620v1.abstract" target="_blank" rel="noopener">here</a>.</p>
<p>To run those scripts, basic knowledge of shell scripting is needed. Good tutorials can be <a href="https://github.com/berkeley-scf/tutorial-using-bash" target="_blank" rel="noopener">found here</a> or <a href="https://datacarpentry.org/shell-genomics/" target="_blank" rel="noopener">here</a>.</p>
<p>Also, sequencing data can be stored into many different formats. <a href="https://binf.snipcademy.com/lessons/sequence-file-formats" target="_blank" rel="noopener">This page</a> is a very complete overview of the types of format and list tools to convert from one to another.</p>
<h2 id="pre-processed-data">Pre-processed data.</h2>
<p>Many datasets have been stored in a clean format in the <a href="https://jhubiostatistics.shinyapps.io/recount/" target="_blank" rel="noopener">recount2 project</a>. This is an excellent place to start if you want to explore algorithms on real world and diverse datasets.</p>
<p>The <a href="http://imlspenticton.uzh.ch:3838/conquer/" target="_blank" rel="noopener">conquer project</a> has a shorter list of very well processed datasets, using <a href="https://www.nature.com/articles/nmeth.4197" target="_blank" rel="noopener">Salmon</a>.</p>
<p><a href="https://bioconductor.org/" target="_blank" rel="noopener">The Bioconductor Project</a> also has a lot of datasets packages.</p>
<h1 id="Stat">Tools needed for analysis</h1>
<p>Many excellent books exist to give a clear understanding of the many tools and concept needed. Freedman&rsquo;s book, <a href="https://www.cambridge.org/core/books/statistical-models/15F12FD4EB5FAD43B85FA824286DD44F" target="_blank" rel="noopener">Statistical Models: Theory and Practice</a> is a must-read, which I consider maybe closer to a philosophical treaty. More applied, Susan&rsquo;s Holmes new book, <a href="http://web.stanford.edu/class/bios221/" target="_blank" rel="noopener">Modern Statistics for Modern Biology</a> is quite complete for tools used in genomics. Key concepts include dimensionality reduction, clustering and modelling. A deeper and mathier book is the <a href="https://web.stanford.edu/~hastie/ElemStatLearn/" target="_blank" rel="noopener">The Elements of  Statistical Learning</a> by Hastie, Tibshirani &amp; Friedman.</p>
<p>A central concept in high-dimensional statistics that characterize genomics is multiple hypothesis testing. <a href="https://statweb.stanford.edu/~ckirby/brad/LSI/monograph_CUP.pdf?fbclid=IwAR2sggb2cAkJmElvE6uz61egXYWu94-4933X1NbOn05AqPFVAeliQFE15PQ" target="_blank" rel="noopener">Efron (2010)</a> is a good reference with a clear Baysien justification. <a href="https://www.pnas.org/content/100/16/9440.abstract" target="_blank" rel="noopener">This paper</a> present the concepts in a genomic setting.</p>
<p>The R language is used in many RNA-Seq applications. To improve your coding practices, I also recommend <a href="http://adv-r.had.co.nz/" target="_blank" rel="noopener">advanced R</a> by Hadley Wickham. The <a href="http://adv-r.had.co.nz/OO-essentials.html#s4" target="_blank" rel="noopener">S4 section</a> of the book will be of particular interest, given that Bioconductor strongly recommands using S4 objects and functions.</p>
<h1 id="Biostat">Applications to RNA-Seq</h1>
<p>We&rsquo;re finally getting to the RNA-Seq analysis part. To start, I advice to begin with a practical tutorial. The <a href="https://bioconductor.github.io/BiocWorkshops/" target="_blank" rel="noopener">Bioconductor 2018 Workshop</a> provides a few that I recommend. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5558107/" target="_blank" rel="noopener">This other one</a> is longer but more complete.
This <a href="https://hectorrdb.github.io/NationalAssembly/" target="_blank" rel="noopener">other tutorial</a> that I wrote is shorter but can be a good place to start.</p>
<p>Once you have seen how to proceed, this <a href="https://peerj.com/preprints/27283/?fbclid=IwAR0WH2eTl2Hwig9xxnt_DuYuIsFORwtRMrcI-LgZ6AujxPX1tsPb6t_esXk" target="_blank" rel="noopener">recent review by Koen, et al.</a> is a good follow-up that very nicely recoup many concepts from above and can point you to further resources.</p>
<p>Then, I recommend to dive into method papers. This list is partly inspired from the <a href="https://github.com/jtleek/genomicspapers" target="_blank" rel="noopener">Leek group paper</a>.</p>
<ul>
<li><a href="http://biostat.jhsph.edu/~jleek/papers/seqvar.pdf" target="_blank" rel="noopener">Sequencing technology does not eliminate biological variability</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/pubmed/16646809" target="_blank" rel="noopener">The first paper behind the Limma package</a> and the <a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2014-15-2-r29" target="_blank" rel="noopener">second, with voom</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2796818/" target="_blank" rel="noopener">edgeR</a></li>
<li><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-11-94" target="_blank" rel="noopener">Key need for Normalization</a>, especially because of <a href="https://www.nature.com/articles/nrg2825" target="_blank" rel="noopener">batch effects</a></li>
<li>Methods for normalization include <a href="https://www.biorxiv.org/content/10.1101/235382v2" target="_blank" rel="noopener">scone</a>, <a href="https://www.nature.com/articles/s41467-017-02554-5" target="_blank" rel="noopener">zinbWave</a> and <a href="https://www.nature.com/articles/s41592-018-0229-2" target="_blank" rel="noopener">scVI</a>.</li>
<li><a href="https://satijalab.org/seurat/" target="_blank" rel="noopener">Seurat</a> is a global framework of analysis</li>
<li>Clustering is a key step of RNA-Seq analysis. <a href="https://f1000research.com/articles/7-1141/v2" target="_blank" rel="noopener">This paper</a> contrasts several methods on real and synthetic data.</li>
<li><a href="https://www.nature.com/articles/s41587-019-0071-9" target="_blank" rel="noopener">This paper</a> proposes a nice overview of trajectory inference</li>
<li><a href="https://www.ncbi.nlm.nih.gov/pubmed/20048385" target="_blank" rel="noopener">Gene Set Enrichment Analysis</a></li>
</ul>
<h1 id="Repro">Reproducible research</h1>
<p>All the analyses you conduct should aim for reproducibility in the science. This <a href="https://www.nature.com/collections/prbfkwmwvz" target="_blank" rel="noopener">nature collection</a> group of papers gives a broad overview of the issue of reproducibility, and of how to fix it.  A first step to try and limit the crisis of reproducibilty in science - and one where (bio)-statisticians have a key role - is statistical and computational reproducibility.</p>
<p>To help, several tool exist. Git and Github are powerful tools and many good tutorial exist (see <a href="https://bids.github.io/2019-01-17-bids/" target="_blank" rel="noopener">here</a> for example). Good coding practices are also necessary. Saving your software in packages is a part of it, and <a href="http://r-pkgs.had.co.nz/" target="_blank" rel="noopener">this book</a> is a good reference on how to build them in an efficient fashion. I also recommend using tools like <a href="https://rmarkdown.rstudio.com/" target="_blank" rel="noopener">RMarkdown</a> or <a href="https://yihui.name/knitr/" target="_blank" rel="noopener">knitr</a> to combine text, math, code and figures in one coherent place when writing papers, reports or even for day-to-day analyses.</p>
<p>While code is now commonly saved and made public for reproducibility purposes, data is unfortunatly not always available. Raw files can be made accessible through ncbi. Pre-processed data (a count matrix for example) can also be made public using a data R package, such as the experiment packages that can be found in <a href="https://www.bioconductor.org/packages/release/BiocViews.html#___ExperimentData" target="_blank" rel="noopener">Bioconductor</a>.</p>

    </div>

    






<div class="article-tags">
  
  <a class="badge badge-light" href="/tag/tutorials/">Tutorials</a>
  
  <a class="badge badge-light" href="/tag/research/">Research</a>
  
</div>













  
  



  
  
  
    
  
  
  
  <div class="media author-card content-widget-hr">
    
      
      <a href="http://hector-rdb.org/"><img class="avatar mr-3 avatar-circle" src="/author/hector-roux-de-bezieux/avatar_hu27f5da07ca72334cd07c72d82138a3f4_397410_270x270_fill_q75_lanczos_center.jpg" alt="Hector Roux de Bézieux"></a>
    

    <div class="media-body">
      <h5 class="card-title"><a href="http://hector-rdb.org/">Hector Roux de Bézieux</a></h5>
      <h6 class="card-subtitle">Ph.D Student in Biostatistics</h6>
      <p class="card-text">Biostatistics Ph.D Student with strong interest in anything &lsquo;omics related.</p>
      <ul class="network-icon" aria-hidden="true">
  
</ul>

    </div>
  </div>
















  
  
  <div class="article-widget content-widget-hr">
    <h3>Related</h3>
    <ul>
      
      <li><a href="/project/caldera/">Caldera</a></li>
      
      <li><a href="/project/dune/">Dune</a></li>
      
      <li><a href="/project/nationalassemby/">ESCAPE</a></li>
      
      <li><a href="/project/tradeseq/">tradeSeq</a></li>
      
    </ul>
  </div>
  





    <div class="project-related-pages content-widget-hr">
      
      

      
      
      

      
      
      

      
      
      
    </div>
  </div>
</article>
  </div>

  <div class="page-footer">
    
    
    <div class="container">
      <footer class="site-footer">
  

  <p class="powered-by">
    
  </p>

  
  






  <p class="powered-by">
    
    
    
    Published with
    <a href="https://wowchemy.com/?utm_campaign=poweredby" target="_blank" rel="noopener">Wowchemy</a>  —
    the free, <a href="https://github.com/wowchemy/wowchemy-hugo-modules" target="_blank" rel="noopener">
    open source</a> website builder that empowers creators.
    
  </p>
</footer>

    </div>
    
  </div>

  
<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h5 class="modal-title">Cite</h5>
        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body">
        <pre><code class="tex hljs"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-outline-primary my-1 js-copy-cite" href="#" target="_blank">
          <i class="fas fa-copy"></i> Copy
        </a>
        <a class="btn btn-outline-primary my-1 js-download-cite" href="#" target="_blank">
          <i class="fas fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

      

    
    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha256-9/aliU8dGd2tb6OSsuzixeV4y/faTqgFtohetphbbj0=" crossorigin="anonymous"></script>

      
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.4/imagesloaded.pkgd.min.js" integrity="sha256-lqvxZrPLtfffUl2G/e7szqSvPBILGbwmsGE1MKlOi0Q=" crossorigin="anonymous"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.6/isotope.pkgd.min.js" integrity="sha256-CBrpuqrMhXwcLLUd5tvQ4euBHCdh7wGlDfNz8vbu/iI=" crossorigin="anonymous"></script>
      

      <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.5.7/jquery.fancybox.min.js" integrity="sha256-yt2kYMy0w8AbtF89WXb2P1rfjcP/HTHLT7097U8Y5b8=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/instant.page/5.1.0/instantpage.min.js" integrity="sha512-1+qUtKoh9XZW7j+6LhRMAyOrgSQKenQ4mluTR+cvxXjP1Z54RxZuzstR/H9kgPXQsVB8IW7DMDFUJpzLjvhGSQ==" crossorigin="anonymous"></script>

      

      
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.2.0/highlight.min.js" integrity="sha512-TDKKr+IvoqZnPzc3l35hdjpHD0m+b2EC2SrLEgKDRWpxf2rFCxemkgvJ5kfU48ip+Y+m2XVKyOCD85ybtlZDmw==" crossorigin="anonymous"></script>
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.2.0/languages/r.min.js"></script>
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.2.0/languages/latex.min.js"></script>
        
      

    

    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.7.1/leaflet.min.js" integrity="sha512-SeiQaaDh73yrb56sTW/RgVdi/mMqNeM2oBwubFHagc5BkixSpP1fvqF47mKzPGWYSSy4RwbBunrJBQ4Co8fRWA==" crossorigin="anonymous"></script>
    

    
    
    <script>const code_highlighting = true;</script>
    

    
    
    
    
    
    
    <script>
      const search_config = {"indexURI":"/index.json","minLength":1,"threshold":0.3};
      const i18n = {"no_results":"No results found","placeholder":"Search...","results":"results found"};
      const content_type = {
        'post': "Posts",
        'project': "Projects",
        'publication' : "Publications",
        'event' : "Events",
        'slides' : "Slides"
        };
    </script>
    

    
    

    
    
    <script id="search-hit-fuse-template" type="text/x-template">
      <div class="search-hit" id="summary-{{key}}">
      <div class="search-hit-content">
        <div class="search-hit-name">
          <a href="{{relpermalink}}">{{title}}</a>
          <div class="article-metadata search-hit-type">{{type}}</div>
          <p class="search-hit-description">{{snippet}}</p>
        </div>
      </div>
      </div>
    </script>
    

    
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/3.2.1/fuse.min.js" integrity="sha256-VzgmKYmhsGNNN4Ph1kMW+BjoYJM2jV5i4IlFoeZA9XI=" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>
    

    
    

    
    
    

    
    

    
    

    
    
    
    
    
    
    
    
    
    
      
      
    
    
    <script src="/js/wowchemy.min.572b7caf7fe601716a0f74d67d754c21.js"></script>

    






</body>
</html>
